video/decode/vd_lavc.c

```diff
@@ -568,6 +572,19 @@ static void update_image_params(struct dec_video *vd, AVFrame *frame,
     vd_ffmpeg_ctx *ctx = vd->priv;
     struct MPOpts *opts = ctx->opts;
 
+#if HAVE_AVUTIL_MASTERING_METADATA
+    // Get the reference peak (for HDR) if available. This is cached into ctx
+    // when it's found, since it's not available on every frame (and seems to
+    // be only available for keyframes)
+    AVFrameSideData *sd = av_frame_get_side_data(frame,
+                          AV_FRAME_DATA_MASTERING_DISPLAY_METADATA);
+    if (sd) {
+        AVMasteringDisplayMetadata *mdm = (AVMasteringDisplayMetadata *)sd->data;
+        if (mdm->has_luminance)
+            ctx->cached_hdr_peak = av_q2d(mdm->max_luminance);
```

Comment:
Looked at libavcodec. Seems like currently, the denominator is always set to non-0, so I guess we're actually safe currently.

There's also code to read this in the Matroska demuxer. It reads floats directly from the file and turns them into integer fractions with a fixed denominator. Sure seems pointless.

Btw. this code doesn't catch the equivalent "per-stream" demuxer side data yet.